src/frontend/js/components/DashBoard/Navigator/index.tsx

```diff
@@ -0,0 +1,48 @@
+import { NavLink, useRoutes } from 'react-router-dom';
+import { ReactNode } from 'react';
+import { defineMessages, FormattedMessage } from 'react-intl';
+import useDashboardRoutes from '../hooks/useDashboardRoutes';
+import { Spinner } from '../../Spinner';
+
+export const messages = defineMessages({
+  loadingNavigator: {
+    id: 'components.Dashboard.Navigator.loadingInitial',
+    defaultMessage: 'Loading navigator',
+    description:
+      'Accessible text for the initial loading spinner displayed when navigator is fetching',
+  },
+});
+
+const DashBoardNavigator = () => {
+  const { intlDashboardRoutes } = useDashboardRoutes();
+  const routesNode = useRoutes(intlDashboardRoutes);
+
+  if (!intlDashboardRoutes) {
+    return (
+      <Spinner aria-labelledby="loading-routes">
+        <span id="loading-course">
+          <FormattedMessage {...messages.loadingNavigator} />
+        </span>
+      </Spinner>
+    );
+  }
+
+  return (
+    <>
+      <nav>
+        {intlDashboardRoutes
+          .filter((route) => route.path && route.title)
+          .map<ReactNode>((route, index): ReactNode => {
+            return (
+              <NavLink to={route.path as string} key={`nav_${index}`}>
+                {route.title}
+              </NavLink>
+            );
```

Comment:
Improve typing, reduce callstack and do not use array index as component key.
```suggestion
          .map(
          (route) =>
            route.path &&
            route.title && (
              <NavLink to={route.path} key={`nav_${route.path}`}>
                {route.title}
              </NavLink>
            ),
        )
```